From 1d284885de20292b0db4cfb175d3edd610c558f8 Mon Sep 17 00:00:00 2001
From: Ilan Peer <ilan.peer@intel.com>
Date: Tue, 23 Apr 2024 10:47:45 +0530
Subject: [PATCH] UPSTREAM: nl80211: Do not allow off channel when frequency
 is not specified

Do not allow offchannel operation for action frame transmission if
no frequency is specified, as this doesn't make sense.

(Cherry-picked from
https://w1.fi/cgit/hostap/commit/?id=d8cae2d02449e35e5fb976fd1385071c2122152e)

Signed-off-by: Ilan Peer <ilan.peer@intel.com>
Signed-off-by: Yuvarani V <quic_yuvarani@quicinc.com>
---
 src/drivers/driver_nl80211.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index feb5005..23838a9 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -9337,6 +9337,9 @@ static int wpa_driver_nl80211_send_action(struct i802_bss *bss,
 	    bss->flink->beacon_set)
 		offchanok = 0;
 
+	if (!freq && is_sta_interface(drv->nlmode))
+		offchanok = 0;
+
 	wpa_printf(MSG_DEBUG, "nl80211: Send Action frame (ifindex=%d, "
 		   "freq=%u MHz wait=%d ms no_cck=%d offchanok=%d)",
 		   drv->ifindex, freq, wait_time, no_cck, offchanok);
-- 
2.17.1

